%YAML:1.0
---
################### System ###################
start_time: 1642516757.125902

R0: !!opencv-matrix
   rows: 3
   cols: 3
   dt: d
   data: [-0.00883249, -0.99959381, 0.02709611, 
         0.92718465, 0.00196163, 0.37459948,
         -0.37450048, 0.02843174, 0.92679072 ]
         
V0: !!opencv-matrix
   rows: 3
   cols: 1
   dt: d
   data: [0.0, 0.0, 0.0]

t0: !!opencv-matrix
   rows: 3
   cols: 1
   dt: d
   data: [0.08133085, -0.12753305, 0.06595351]



cloud_path: "/home/yuanicg/code/CANNY_EVIT/cloud/Sofa_fast.txt"
result_path: "/home/yuanicg/data/EVIT/result/sofa_fast_result/"
timeSurface_decay_factor: 20.0e-3
imu_num_for_frame: 7
window_size: 6

frame_num_for_init: 5
imu_num_for_init_frame: 5
init_freq: 30 # hz

min_num_events_for_frame: 30000
min_num_imu_for_frame: 4

################### Event Camera ###################
height: 480
width: 640
distortion_model: "plumb_bob" # equidistant
K: !!opencv-matrix
   rows: 3
   cols: 3
   dt: d
   data: [ 327.32749,    0.     ,  304.97749,
         0.     ,  327.46184,  235.37621,
         0.     ,    0.     ,    1.     ]
distortion: !!opencv-matrix
   rows: 5
   cols: 1
   dt: d
   data: [-0.031982, 0.041966, -0.000507, -0.001031, 0.000000]

P: !!opencv-matrix
   rows: 3
   cols: 4
   dt: d
   data: [ 339.69174,    0.     ,  305.8753 ,    0.     ,
            0.     ,  340.96127,  235.33929,    0.     ,
            0.     ,    0.     ,    1.     ,    0.     ]

Rbc: !!opencv-matrix
   rows: 3
   cols: 3
   dt: d
   data: [0.01120354,  0.0218441 ,  0.99969861,
         -0.99976045, -0.01855423,  0.01160966,  
          0.01880224, -0.9995892 ,  0.021631]
tbc: !!opencv-matrix
   rows: 3
   cols: 1
   dt: d
   data: [0.12907465,  0.07967281, 0.025]



################### imu parameter ###################
acc_n: 0.12655720309610252         # accelerometer measurement noise standard deviation.
gyr_n: 0.07294729852113113         # gyroscope measurement noise standard deviation.
acc_w: 5.6386016813618435e-03      # accelerometer bias random work noise standard deviation.
gyr_w: 6.996094830870257e-04       # gyroscope bias random work noise standard deviation.
g_norm: 9.81     # gravity magnitude


################### optimizer ###################
patch_size_X: 3
patch_size_Y: 3
field_type: "inv_time_surface" # "distance_field" or "inv_time_surface"
polarity_prediction: 0 #bool
batch_size: 300
max_registration_point: 4000


################### GLOG ###################
log_dir: ""
log_color: 1
log_also_to_stderr: 1



################### Viewer ###################
point_size: 1
frame_size: 0.02
